test/ergw_http_api_SUITE.erl

```diff
@@ -127,6 +142,107 @@ end_per_suite(Config) ->
+    Session2 = maps:get(<<"Session">>, Res2),
+    ?match([], Session2),
+    ?match(404, StatusCode),
+
+    delete_pdp_context(S, GtpC1),
+    ok = meck:wait(ggsn_gn, terminate, '_', 2000),
+    meck_validate(Config),
+    ok.
+
+http_api_sessions_list() ->
+    [{doc, "Check /session API"}].
+http_api_sessions_list(Config) ->
+    S = make_gtp_socket(Config),
+    {GtpC1, _, _} = create_pdp_context(S),
+    {GtpC2, _, _} = create_pdp_context(random, S, GtpC1),
+
+    URL1 = get_test_url("/api/v1/sessions"),
+    {ok, {_, _, Body1}} = httpc:request(get, {URL1, []},
+                                        [], [{body_format, binary}]),
+    Res1 = jsx:decode(Body1, [return_maps]),
+    TotalCount = maps:get(<<"TotalCount">>, Res1),
+    ?equal(2, TotalCount),
+    Sessions = maps:get(<<"Sessions">>, Res1),
+    lists:foreach(fun (Session) ->
+                          MSIDN = maps:get(<<"MSIDN">>, Session),
+                          ?match(?MSISDN, MSIDN),
+                          RemoteControlIp = maps:get(<<"RemoteControlIp">>, Session),
+                          ?match(<<"127.127.127.127">>, RemoteControlIp),
+                          RemoteDataIp = maps:get(<<"RemoteDataIp">>, Session),
+                          ?match(<<"127.127.127.127">>, RemoteDataIp),
+                          Version = maps:get(<<"Version">>, Session),
+                          ?match(<<"v1">>, Version)
+                  end, Sessions),
+    delete_pdp_context(S, GtpC1),
+    delete_pdp_context(S, GtpC2),
+    ok = meck:wait(ggsn_gn, terminate, '_', 2000),
+    meck_validate(Config),
+    ok.
+
+http_api_session_stop() ->
+    [{doc, "Check DELETE /session/id/bearer_id"}].
+http_api_session_stop(Config) ->
+    % Create session and request it
+    S = make_gtp_socket(Config),
+    {_GtpC1, _, _} = create_pdp_context(S),
+
+    URL1 = get_test_url("/api/v1/sessions"),
+    {ok, {_, _, Body1}} = httpc:request(get, {URL1, []},
+                                        [], [{body_format, binary}]),
+    Res1 = jsx:decode(Body1, [return_maps]),
+    TotalCount1 = maps:get(<<"TotalCount">>, Res1),
+    ?equal(1, TotalCount1),
+
+    % Stop session
+    URL2 = get_test_url("/api/v1/sessions/111111111111111/5"),
+    {ok, {_, _, Body2}} = httpc:request(delete, {URL2, [], "", []},
+                                        [], [{body_format, binary}]),
+    Res2 = jsx:decode(Body2, [return_maps]),
+    Result2 = maps:get(<<"Result">>, Res2),
+    ?match(<<"ok">>, Result2),
+
+    % Stop non-existent session
+    URL3 = get_test_url("/api/v1/sessions/111111111111112/5"),
+    {ok, {{_, StatusCode, _}, _, Body3}} = httpc:request(delete, {URL3, [], "", []},
+                                                         [], [{body_format, binary}]),
+    Res3 = jsx:decode(Body3, [return_maps]),
+    Result3 = maps:get(<<"Result">>, Res3),
+    ?match(<<"Session not found">>, Result3),
+    ?match(404, StatusCode),
+
+    % Check that session was destroyed after stop
```

Comment:
you need to wait for the context to be terminated, otherwise this could produce random failures. To make sure all context are gone use ```wait4tunnels(?TIMEOUT)``` (e.g.: https://github.com/travelping/ergw/blob/master/test/pgw_SUITE.erl#L337)